modules/twinklefluff.js

```diff
@@ -76,124 +77,91 @@ Twinkle.fluff = {
+				var oldrev, onode;
+				if (document.getElementById('mw-revision-info')) { // old revision
+					oldrev = Morebits.queryString.get('oldid');
+					onode = 'mw-revision-info';
+				} else { // diff
+					var old_rev_url = $("#mw-diff-otitle1").find("strong a").attr("href");
+					oldrev = new Morebits.queryString( old_rev_url.split( '?', 2 )[1] ).get('oldid');
+					onode = 'mw-diff-otitle1';
 				}
-				otitle = otitle1.parentNode;
-				ntitle = ntitle1.parentNode;
-			} catch( e ) {
-				// no old, nor new title, nothing to do really, return;
-				return;
-			}
-
-			var old_rev_url = $("#mw-diff-otitle1").find("strong a").attr("href");
-
-			// Lets first add a [edit this revision] link
-			var query = new Morebits.queryString( old_rev_url.split( '?', 2 )[1] );
-
-			var oldrev = query.get('oldid');
-
-			var revertToRevision = document.createElement('div');
-			revertToRevision.setAttribute( 'id', 'tw-revert-to-orevision' );
-			revertToRevision.style.fontWeight = 'bold';
-
-			var revertToRevisionLink = revertToRevision.appendChild( document.createElement('a') );
-			revertToRevisionLink.href = "#";
-			$(revertToRevisionLink).click(function(){
-				Twinkle.fluff.revertToRevision(oldrev);
-			});
-			revertToRevisionLink.appendChild( spanTag( 'Black', '[' ) );
-			revertToRevisionLink.appendChild( spanTag( 'SaddleBrown', 'restore this version' ) );
-			revertToRevisionLink.appendChild( spanTag( 'Black', ']' ) );
-
-			otitle.insertBefore( revertToRevision, otitle.firstChild );
 
-			if( document.getElementById('differences-nextlink') ) {
-				// Not latest revision
-				var new_rev_url = $("#mw-diff-ntitle1").find("strong a").attr("href");
-				query = new Morebits.queryString( new_rev_url.split( '?', 2 )[1] );
-				var newrev = query.get('oldid');
-				revertToRevision = document.createElement('div');
-				revertToRevision.setAttribute( 'id', 'tw-revert-to-nrevision' );
+				var revertToRevision = document.createElement('div');
+				revertToRevision.setAttribute( 'id', 'tw-revert-to-orevision' );
 				revertToRevision.style.fontWeight = 'bold';
-				revertToRevisionLink = revertToRevision.appendChild( document.createElement('a') );
+
+				var revertToRevisionLink = buildLink('SaddleBrown', 'restore this version');
 				revertToRevisionLink.href = "#";
 				$(revertToRevisionLink).click(function(){
-					Twinkle.fluff.revertToRevision(newrev);
+					Twinkle.fluff.revertToRevision(oldrev);
 				});
-				revertToRevisionLink.appendChild( spanTag( 'Black', '[' ) );
-				revertToRevisionLink.appendChild( spanTag( 'SaddleBrown', 'restore this version' ) );
-				revertToRevisionLink.appendChild( spanTag( 'Black', ']' ) );
-				ntitle.insertBefore( revertToRevision, ntitle.firstChild );
+				revertToRevision.appendChild(revertToRevisionLink);
 
-				return;
+				var otitle = document.getElementById(onode).parentNode;
+				otitle.insertBefore( revertToRevision, otitle.firstChild );
```

Comment:
Nitpick -- I don't really care about `( code )` vs. `(code)` but you probably didn't mean to put `(code )` (with the trailing space). One day we should add some linter rules and unify all this. I think there are utilities to automatically update the code, too.